tests/test_permissions.py

```diff
@@ -170,42 +168,61 @@ def test_permissions_denied_process(factory, client_1, client_2):
 
 @pytest.mark.remote_only  # no check on permissions with the local backend
 @pytest.mark.slow
-@pytest.mark.xfail(reason='permission check not yet implemented in the backend')
-def test_permissions_denied_model_process(factory, client_1, client_2, network):
-    # setup
-
+@pytest.mark.parametrize('client_1_permissions,client_2_permissions,expected_success', [
+    (
+        Permissions(public=False, authorized_ids=[]),
+        Permissions(public=False, authorized_ids=[]),
+        False
+    ),
+    (
+        Permissions(public=False, authorized_ids=[MSP_IDS[1]]),
+        Permissions(public=False, authorized_ids=[]),
+        True
+    ),
+])
+def test_permissions_model_process(
+    client_1_permissions,
+    client_2_permissions,
+    expected_success,
+    factory,
+    client_1,
+    client_2,
+    network
+):
+    """Test that a traintuple can/cannot process an in-model depending on permissions."""
     datasets = []
     algos = []
     for client in network.clients[:2]:
+        permissions = client_1_permissions if client == client_1 else client_2_permissions
```

Comment:
We could handle that in the for loop directly:

```python
for client, permissions in zip([client_1, client_2], [client_1_permissions, client_2_permissions]):
```

Reply:
Done :)